JamJar/StitchedJamJarAVPlayerViewController.swift

```diff
@@ -0,0 +1,253 @@
+    }
+    
+    // This controller needs to remove JamJar unique elements
+    override func removeObservers() {
+        super.removeObservers()
+        
+        //Remove buttons to avoid duplication if the view is reloading
+        self.rewindButton.removeFromSuperview()
+        self.fastFowardButton.removeFromSuperview()
+        
+        // Remove observer for checking if video ended
+        NSNotificationCenter.defaultCenter().removeObserver(self, name: AVPlayerItemDidPlayToEndTimeNotification, object: self.player?.currentItem)
+    }
+    
+    override func didReceiveMemoryWarning() {
+        super.didReceiveMemoryWarning()
+        // Dispose of any resources that can be recreated.
+    }
+    
+    // Functions to add custom UI Elements
+    
+    // Rewind Button Button
+    func createRewindButton() {
+        self.rewindButton = UIButton(type: UIButtonType.RoundedRect) as UIButton
+        rewindButton.frame = CGRectMake(10, 0, 30, 30)
+        rewindButton.tintColor = UIColor.whiteColor()
+        rewindButton.setImage(self.imageFromSystemBarButton(UIBarButtonSystemItem.Rewind), forState: .Normal)
+        rewindButton.addTarget(self, action: #selector(StitchedJamJarAVPlayerViewController.rewindButtonAction(_:)), forControlEvents: UIControlEvents.TouchUpInside)
+        
+        self.bottomBar.addSubview(rewindButton)
+    }
+    
+    // Fast Forward Button
+    func createFastForwardButton() {
+        self.fastFowardButton = UIButton(type: UIButtonType.RoundedRect) as UIButton
+        fastFowardButton.frame = CGRectMake(70, 0, 30, 30)
+        fastFowardButton.tintColor = UIColor.whiteColor()
+        fastFowardButton.setImage(self.imageFromSystemBarButton(UIBarButtonSystemItem.FastForward), forState: .Normal)
+        fastFowardButton.addTarget(self, action: #selector(StitchedJamJarAVPlayerViewController.fastForwardButtonAction(_:)), forControlEvents: UIControlEvents.TouchUpInside)
+        
+        self.bottomBar.addSubview(fastFowardButton)
+    }
+    
+    func rewindButtonAction(sender:UIButton!)
+    {
+        if !self.overlappingVideos.isEmpty {
+            let newVideoId = self.storedAVPlayers.last?.videoId
+            changeCurrentVideo(newVideoId!)
+        }
+    }
+    
+    func fastForwardButtonAction(sender:UIButton!)
+    {
+        if !self.overlappingVideos.isEmpty {
+            let newVideoId = self.storedAVPlayers.first?.videoId
+            changeCurrentVideo(newVideoId!)
+        }
+    }
+    
+    // check for new overlaps
+    override func updateTimeLabel(elapsedTime: Float64, duration: Float64) {
+        super.updateTimeLabel(elapsedTime, duration: duration)
+        
+        //TODO: find overlaps
+        let edges = jamjar.nodes![String(self.currentVideo.id!)]
```

Comment:
Uhhh, this seems like a **lot** of computation to do every time this callback is called.  I feel like there's gotta be some preprocessing that we can do on load here so that we can do a constant-time lookup for any edges.